packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -1144,5 +1194,151 @@ foo?.bar(/* comment */a,
+            },
+          ],
+        },
+      ],
+    })),
+    // case with this keyword at the start of expression
+    {
+      code: '!this.bar || !this.bar.baz;',
+      output: null,
+      errors: [
+        {
+          messageId: 'preferOptionalChain',
+          suggestions: [
+            {
+              messageId: 'optionalChainSuggest',
+              output: '!this.bar?.baz;',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: '!a.b || !a.b();',
+      output: null,
+      errors: [
+        {
+          messageId: 'preferOptionalChain',
+          suggestions: [
+            {
+              messageId: 'optionalChainSuggest',
+              output: '!a.b?.();',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: '!foo.bar || !foo.bar.baz;',
+      output: null,
+      errors: [
+        {
+          messageId: 'preferOptionalChain',
+          suggestions: [
+            {
+              messageId: 'optionalChainSuggest',
+              output: '!foo.bar?.baz;',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: '!foo[bar] || !foo[bar]?.[baz];',
+      output: null,
+      errors: [
+        {
+          messageId: 'preferOptionalChain',
+          suggestions: [
+            {
+              messageId: 'optionalChainSuggest',
+              output: '!foo[bar]?.[baz];',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: '!foo || !foo?.bar.baz;',
+      output: null,
+      errors: [
+        {
+          messageId: 'preferOptionalChain',
+          suggestions: [
+            {
+              messageId: 'optionalChainSuggest',
+              output: '!foo?.bar.baz;',
+            },
+          ],
+        },
+      ],
+    },
+    // two  errors
+    {
+      code: noFormat`(!foo || !foo.bar || !foo.bar.baz) && (!baz || !baz.bar || !baz.bar.foo);`,
+      output: null,
+      errors: [
+        {
+          messageId: 'preferOptionalChain',
+          suggestions: [
+            {
+              messageId: 'optionalChainSuggest',
+              output: noFormat`(!foo?.bar?.baz) && (!baz || !baz.bar || !baz.bar.foo);`,
+            },
+          ],
+        },
+        {
+          messageId: 'preferOptionalChain',
+          suggestions: [
+            {
+              messageId: 'optionalChainSuggest',
+              output: noFormat`(!foo || !foo.bar || !foo.bar.baz) && (!baz?.bar?.foo);`,
+            },
+          ],
+        },
+      ],
+    },
+    // TODO: deepest left node already pre-optional chained
+    // {
+    //   code: '!foo?.bar || !foo?.bar.baz;',
+    //   output: null,
+    //   errors: [
+    //     {
+    //       messageId: 'preferOptionalChain',
+    //       suggestions: [
+    //         {
+    //           messageId: 'optionalChainSuggest',
+    //           output: '!foo?.bar?.baz;',
+    //         },
+    //       ],
+    //     },
+    //   ],
+    // },
```

Comment:
Again, what's the status here?

We generally don't check in `TODO` code if possible (it gets hard to keep track of this stuff). I'd rather this be a followup issue that gets filed.

Reply:
Opened an issue instead (#5697)